python/mlx/nn/layers/positional_encoding.py

```diff
@@ -25,20 +25,35 @@ class RoPE(Module):
         base (float, optional): The base used to compute angular frequency for
             each dimension in the positional encodings. Default: ``10000``.
         scale (float, optional): The scale used to scale the positions. Default: ``1.0``.
+        max_position (int, optional): The maximum position for which to precompute the sin and cos values. Default: 2048.
+        dtype: Data type of the computation. Default: mx.float32.
+
+    Attributes:
+        _cos_sin_theta_key (tuple): Cached key for the precomputed cosine and sine values.
+        _cos_sin_theta_value (tuple): Cached cosine and sine values.
+        _freq (mx.ndarray): Frequency terms used in computing theta values.
     """
 
+    _cos_sin_theta_key = None
+    _cos_sin_theta_value = None
+    _freq = None
+
     def __init__(
         self,
         dims: int,
         traditional: bool = False,
         base: float = 10000,
         scale: float = 1.0,
+        max_position: int = 2048,
```

Comment:
To enable building the theta during initialization, we need to know the maximum position and data type.